analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverloadsShouldBeGrouped.cs

```diff
@@ -51,28 +61,13 @@ protected override MemberInfo CreateMemberInfo(SyntaxNodeAnalysisContext c, Memb
             return null;
         }
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(c =>
-            {
-                var typeDeclaration = (TypeDeclarationSyntax)c.Node;
-                CheckMembers(c, typeDeclaration.Members);
-            },
-            SyntaxKind.ClassDeclaration,
-            SyntaxKind.InterfaceDeclaration,
-            SyntaxKind.StructDeclaration);
-        }
-
-        private static bool IsValidMemberForOverload(MemberDeclarationSyntax member)
-        {
-            if (member is MethodDeclarationSyntax methodDeclaration)
-            {
-                return methodDeclaration.ExplicitInterfaceSpecifier == null;
-            }
-            return true;
-        }
+        protected override IEnumerable<MemberDeclarationSyntax> GetMemberDeclarations(SyntaxNode node) =>
+            ((TypeDeclarationSyntax)node).Members;
 
-        private static bool IsStatic(BaseMethodDeclarationSyntax declaration) => declaration.Modifiers.Any(x => x.Kind() == SyntaxKind.StaticKeyword);
+        private static bool IsValidMemberForOverload(MemberDeclarationSyntax member) =>
+            !(member is MethodDeclarationSyntax methodDeclaration) || methodDeclaration.ExplicitInterfaceSpecifier == null;
 
+        private static bool IsStatic(BaseMethodDeclarationSyntax declaration) =>
```

Comment:
I'm pretty sure I've seen a similar check in another rule. Could you please extract an extension method and move it to `BaseMethodDeclarationSyntaxExtensions`?

Edit: there is one usage also on the vb rule, we could try to extract it to the shared project.

Reply:
I moved it to the extensions. The SyntaxKinds are different so it can not go into the shared project.